src/Shared/EscapingUtilities.cs

```diff
@@ -79,6 +59,22 @@ internal static string UnescapeAll
             StringBuilder unescapedString = StringBuilderCache.Acquire(escapedString.Length);
 
             int currentPosition = 0;
+            int escapedStringLength = escapedString.Length;
+            if (trim)
+            {
+                while (currentPosition < escapedString.Length && Char.IsWhiteSpace(escapedString[currentPosition]))
+                {
+                    currentPosition++;
+                }
+                if (currentPosition == escapedString.Length)
+                {
+                    return String.Empty;
+                }
+                while (Char.IsWhiteSpace(escapedString[escapedStringLength]))
+                {
+                    escapedStringLength--;
+                }
```

Comment:
side-note: #5663 includes a Trim() implementation operating on spans which could potentially be used here.

Reply:
One thing to note is that if `Trim()` does not modify the string, and `ToString()` is done on the span, it _will_ allocate a new string:

https://github.com/Therzok/BenchPlayground/blob/820035919d2a47506841041fe7caea799dc939ea/BenchPlayground/Benchmarks/runtime/SpanToString.md

Reply:
Indeed, the .NET Core implementation of `ReadOnlySpan<char>` does not recover the original string object in `ToString()` even if it spans the entire string. This is handled in #5663 by keeping track of the string reference separately so it can be returned if the span hasn't changed.